libr/bin/format/mach0/mach0.c

```diff
@@ -1227,17 +1226,19 @@ static int parse_dylib(struct MACH0_(obj_t) *bin, ut64 off) {
 
 	ut64 offname = off + dl.dylib.name;
 	if (offname + R_BIN_MACH0_STRING_LENGTH > bin->size) {
+		free (lib);
 		return false;
 	}
 
-	memset (bin->libs[lib], 0, R_BIN_MACH0_STRING_LENGTH);
 	len = r_buf_read_at (bin->b, off + dl.dylib.name,
-		(ut8*)bin->libs[lib], R_BIN_MACH0_STRING_LENGTH - 1);
-	bin->libs[lib][R_BIN_MACH0_STRING_LENGTH - 1] = 0;
+		(ut8*) lib, R_BIN_MACH0_STRING_LENGTH - 1);
 	if (len < 1) {
 		bprintf ("Error: read (dylib str)");
+		free (lib);
 		return false;
 	}
+
+	r_pvector_push (&bin->libs_cache, lib);
```

Comment:
even if vectors are better, i dont think an RList is really a big issue, and having rlist internally also avoids us to rewrite the rest of the callers, so maybe just having the RList internally and not freed later is enough at least until we decide to break abi, but this abi break will require not just rewriting all the callers, but also all the plugins, and thats imho a little out of the scope for this optimization

Reply:
I've seen massive differences in the past in other codebases when switching from list to vector, which is why I also did it here.

We can't fully cache the lists though, because free is overridden in `bobj.c`, leading to double frees..